deeppavlov/models/classifiers/sklearn_component.py

```diff
@@ -0,0 +1,219 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Any, Union
+import numpy as np
+import pickle
+from pathlib import Path
+from scipy.sparse import issparse, csr_matrix
+from scipy.sparse import vstack, hstack
+import inspect
+
+from deeppavlov.core.common.errors import ConfigError
+from deeppavlov.core.common.registry import register, cls_from_str
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.models.estimator import Estimator
+
+log = get_logger(__name__)
+
+
+@register("sklearn_component")
+class SklearnComponent(Estimator):
+    def __init__(self, model_name: str,
+                 save_path: Union[str, Path] = None,
+                 load_path: Union[str, Path] = None,
+                 infer_method: str = "predict", **kwargs) -> None:
+
+        super().__init__(save_path=save_path, load_path=load_path, **kwargs)
+        self.model_name = model_name
+        self.model_params = kwargs
+        self.model = None
+        self.pipe_params = {}
+        for required in ["in", "out", "fit_on", "main", "name"]:
+            self.pipe_params[required] = self.model_params.pop(required, None)
+
+        self.load()
+        self.infer_method = infer_method
+        self.epochs_done = 0
+        self.batches_seen = 0
+        self.train_examples_seen = 0
+
+    def fit(self, *args, **kwargs) -> None:
+        n_inputs = len(self.pipe_params["in"]) if isinstance(self.pipe_params["in"], list) else 1
+        x_features = self.compose_input_data(args[:n_inputs])
+        if len(args) > n_inputs:
+            y_ = np.squeeze(np.array(args[-1]))
+        else:
+            y_ = None
+
+        try:
+            log.info("Fitting model {}".format(self.model_name))
+            self.model.fit(x_features, y_)
+        except TypeError or ValueError:
+            try:
+                if issparse(x_features):
+                    log.info("Converting input for model {} to dense array".format(self.model_name))
+                    self.model.fit(x_features.todense(), y_)
+                else:
+                    log.info("Converting input for model {} to sparse array".format(self.model_name))
+                    self.model.fit(csr_matrix(x_features), y_)
+            except Exception:
+                raise ConfigError("Can not fit on the given data".format(self.model_name))
+
+        return
+
+    def __call__(self, *args, **kwargs) -> np.ndarray:
+        predictions = self.infer_on_batch(args)
```

Comment:
`infer_on_batch` can be renamed in `__call__(self, *x):`

Reply:
This component generally needs to be removed, it is superfluous, sorry I forgot about it.

Reply:
This file was deleted.